### PR TITLE
Lets a caller of Strings.shorten choose the length and the indicator

### PR DESCRIPTION
**Pull request by jakobvogel**
### Description

`Strings.shorten` offered no say over the two things a caller most often wants to decide: how far back it may reach for a word boundary, and what marks the omission. The indicator was hard-coded to three dots — while the very same class reaches for the typographic `…` in `limit`.

```java
// existing — behaviour untouched, parameter renamed only
public static String shorten(String string, int maxUnshortenedLength)

// new — the result never exceeds maxLength, as the indicator's length is reserved
public static String shorten(String string, int maxLength, int maxCutback, String omissionIndicator)
```

Both now delegate to one private `cutBackToWordBoundary`, so there is a single implementation. The existing method passes cut position `maxUnshortenedLength - 1` and `"..."`; the new one passes `maxLength - omissionIndicator.length()`.

**No breaking change — deliberately.** The existing method keeps its exact semantics, including the quirk that its length is the one up to which a string is returned *unchanged*, so a shortened result runs two chars past it (cut at `length - 1`, then three chars of indicator). Callers may well depend on that, so it stays. The new overload gets the honest contract instead of inheriting the quirk.

**The parameter rename.** `numChars` becomes `maxUnshortenedLength`, because that is what the code actually tests (`string.length() <= numChars`), whereas the Javadoc called it *"new maximum length of string"* — which it never was. A parameter name binds neither the source nor the binary contract; its meaning does. So only the name moves, and the Javadoc now states the two-char overrun rather than denying it.

#### A crash, fixed

The loop guarded `numChars > 0` — which never changes inside the loop — where it meant `index > 0`. Two ways to fall off the start of the string:

- a length shorter than the word being cut through, so the backtrack walks `index` to `-1` and `charAt` throws
- a length of zero or negative, so `index` starts negative and `substring` throws before the loop is ever entered

`Strings.shorten("abcdef", 3)` and `Strings.shorten("abc", 0)` both threw `StringIndexOutOfBoundsException`.

#### Verified against the previous implementation

Rather than reason about equivalence, I ran the old implementation and the new one against 300k random inputs (lengths 0–40, limits −3 to 41):

```
compared=259175  identical=259175  mismatches=0  (legacy threw on 40825 inputs, now handled)
```

Identical on every input where it returned at all; the only difference is that it no longer throws.

#### Tests

`shorten` had **none** — which is presumably how both the wrong Javadoc and the crash survived. Seven cases added, following the existing style in `StringsTest`, including one that deliberately pins the two chars the existing length runs over, with a comment saying callers may rely on it, so it is not quietly "corrected" later.

Full suite: 463 tests, 0 failures.

#### Note

This is a public API addition to a foundational library, so the overload is permanent surface. The caller that motivated it (a link preview that wants a hard 300-character cap ending in `…` rather than `...`) lives in memoio and will follow in its own PR, once a release carries this.

### Additional Notes

- This PR fixes or works on following ticket(s): [MIO-6573](https://scireum.myjetbrains.com/youtrack/issue/MIO-6573) — no separate SIRI ticket; this came out of the memoio-side work on link previews.

### Checklist

- [x] Code change has been tested and works locally
- [ ] Code was formatted via IntelliJ and follows SonarLint & [best practices](https://scireum.myjetbrains.com/youtrack/articles/MISC-A-16/CodeStyle-JavaDoc) — **the IntelliJ formatter was run** on both changed files (both report *Formatted well*, and I checked the diff touches nothing outside the method and its Javadoc), but **SonarLint was not**: it cannot run headlessly, so *Analyze VCS Changed Files* still needs a pass in the IDE.
- [ ] Patch Tasks: none required.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/main/java/sirius/kernel/commons/Strings.java [hunk 1]
```diff
@@ -682,29 +692,77 @@ public static boolean containsAllowedHtml(@Nullable String content) {
...
         }
-        if (string.length() <= numChars) {
+        if (string.length() <= maxUnshortenedLength) {
             return string;
         }
-        int index = numChars - 1;
-        int maxCutoff = Math.min(string.length() / 2, 10);
-        while (numChars > 0 && Character.isLetterOrDigit(string.charAt(index)) && maxCutoff > 0) {
+
+        return cutBackToWordBoundary(string, maxUnshortenedLength - 1, DEFAULT_MAX_CUTBACK, DEFAULT_OMISSION_INDICATOR);
```
[the remark this answers is not in the document]
Comment (jakobvogel) [reply src/main/java/sirius/kernel/commons/Strings.java:722]: Good catch — reproduced and fixed in 19dd5969.

```
shorten("abcdef", Integer.MIN_VALUE)          -> THREW  Index 2147483647 out of bounds for length 6
shorten("abcdef", Integer.MIN_VALUE, 10, "…") -> THREW  same
```

`MIN_VALUE - 1` wraps to `MAX_VALUE`, so the `Math.max(0, cutIndex)` clamp meant to catch negatives lets a huge *positive* index through. It is not a regression — the previous implementation threw for every non-positive length — but the PR claimed such lengths no longer throw, and `0` and `-5` did already return a value, so being safe for only some negatives was worse than a clear rule.

Both methods now return early, which also makes the documented "indicator alone" case of the new overload explicit rather than something that emerged from clamping.

On the negative cutback: it behaves as zero (cut where the length lands, without hunting for a boundary), which is sensible, so I documented it rather than clamping — clamping would change nothing observable.

The equivalence run against the previous implementation was extended to include `Integer.MIN_VALUE`, `Integer.MAX_VALUE` and their neighbourhoods, since the original corpus only covered limits −3…41 and never reached the overflow: 212805 inputs identical, zero mismatches, and the 87195 inputs where the old code threw are now handled. Three tests added for the edge cases.

File: src/main/java/sirius/kernel/commons/Strings.java [hunk 2]
```diff
@@ -42,6 +42,16 @@ public class Strings {
      */
     private static final String ELLIPSIS = "…";
 
+    /**
+     * Names the omission indicator used by {@link #shorten(String, int)}.
+     */
+    private static final String DEFAULT_OMISSION_INDICATOR = "...";
+
+    /**
+     * Determines how many chars {@link #shorten(String, int)} may give up to reach a word boundary.
```
Comment (sabieber) [inline src/main/java/sirius/kernel/commons/Strings.java]:
```suggestion
     * Defines how many chars {@link #shorten(String, int)} may give up to reach a word boundary.
```

**Review by sabieber — APPROVED** [review]
